File: new-resource-detector/src/test/java/nl/haploid/resource/detector/detector/MockResourceDetector.java
package nl.haploid.resource.detector.detector;

import nl.haploid.event.AtomChangeEvent;
import nl.haploid.resource.detector.service.ResourceDescriptor;
import org.springframework.stereotype.Component;

import java.util.Collection;
import java.util.List;

@Component
public class MockResourceDetector implements ResourceDetector {

	@Override
	public Collection<String> getAtomTypes() {
		return null;
	}

	@Override
	public List<ResourceDescriptor> detect(final List<AtomChangeEvent> events) {
		return null;
	}
}

File: new-resource-detector/src/test/scala/nl/haploid/octowight/channel/scout/detector/MockResourceDetector.scala
package nl.haploid.octowight.channel.scout.detector

import nl.haploid.octowight.AtomGroup
import nl.haploid.octowight.registry.data.{Atom, ResourceRoot}
import org.springframework.stereotype.Component

object MockResourceDetector {
  val AtomCategory = "john"
  val ResourceCollection = "preston"
}

@Component
class MockResourceDetector extends ResourceDetector {

  override def atomCategories = Set(MockResourceDetector.AtomCategory)

  override def detect(atomGroup: AtomGroup, atomIds: Set[Long]): Set[ResourceRoot] =
    atomIds.map(atomId =>
      new ResourceRoot(
        resourceId = null, resourceCollection = MockResourceDetector.ResourceCollection, version = null,
        root = Atom(id = atomId, group = atomGroup)
      )
    )
}
